scripts/convert-svgs-to-icon-mixins.js

```diff
@@ -8,39 +8,111 @@ const directory = args[2];
 const symbols = /[\r\n%#()<>?\[\\\]^`{|}]/g;
 const svgo = new SVGO();
 
-function encodeSVG(data) {
+function parseColorArguments(data) {
+  const color_count = (data.match(/fill="#[^\s]+"/g) || []).length;
+  let arguments = [];
+
+  if (color_count == 1) {
+    arguments.push('$color');
+  } else {
+    for (var i = 0; i < color_count; i++) {
+      arguments.push(`$color${i + 1}`);
+    }
+  }
+
+  return arguments;
+}
+
+function printBaseMixins(svgs) {
+  console.log('// Base definition mixins:');
+  console.log('//------------------------');
+
+  svgs.forEach((svg) => {
+    console.log(`@mixin vf-icon-${svg.name}(${svg.colors.join(', ')}) {`);
+    console.log(` background-image: url("data:image/svg+xml,${encodeSVG(svg.svg, svg.colors)}");`);
+    console.log('}');
+    console.log('');
+  });
+}
+
+function printPatternMixins(svgs) {
+  console.log('// Pattern includes:');
+  console.log('//------------------');
+
+  svgs.forEach((svg) => {
+    console.log(`@include vf-p-icon-${svg.name};`);
+  });
+
+  console.log('// Pattern mixins:');
+  console.log('//----------------');
+
+  svgs.forEach((svg) => {
+    console.log(`@mixin vf-p-icon-${svg.name} {`);
+    console.log(` .p-icon--${svg.name} {`);
+    console.log('   @extend %icon;');
+    console.log(`   @include vf-icon-${svg.name}(${svg.colors.join(', ')});`);
+    console.log('');
+    console.log("   [class*='--dark'] &,");
+    console.log('   &.is-light {');
+    console.log(`     @include vf-icon-${svg.name}(${svg.colors.join(', ')});`);
+    console.log('   }');
+    console.log(' }');
+    console.log('}');
+    console.log('');
+  });
+
+  console.log('// **Base and Pattern mixins accurate as of December 2020**');
```

Comment:
Nice one thanks, may be worth adding Vanilla version for reference as well.

```suggestion
  console.log('// **Base and Pattern mixins accurate as of December 2020, Vanilla v2.21.0**');
```